tpgtools/templates/resource.go.tmpl

```diff
@@ -474,23 +477,24 @@ func resource{{$.PathType}}Update(d *schema.ResourceData, meta interface{}) erro
 {{- end }}
 
 func resource{{$.PathType}}Delete(d *schema.ResourceData, meta interface{}) error {
+{{- if $.HasDelete }}
```

Comment:
If the resource has no delete method should we attempt to zero-out the field? I'm not sure that skipping delete entirely is what we want to do in this situation.

What does the lack of a delete function mean coming from the DCL? Does that mean the resource is a singleton, fine-grained, or just lacks a standard DELETE call?

Reply:
A resource without a delete method could be a singleton or it could just be a resource that can't be deleted like kms crypto key. I'm not sure what we want to do when delete is called on those resources, but anything that includes `client.Delete{{$.DCLTitle}}` won't compile because that method doesn't exist.

Reply:
Hm, we may need to expand the DCL's idea of "Delete" then. In the provider historically we have tried to have some sense of the "zero" value of a singleton, and when a singleton resource is deleted we try to set it to the zero value rather than just abandoning the resource entirely.

You can see this in how the IAM resources function. When an IAM policy is deleted we set the IAM policy for the resource to an empty policy, even though there is no DeleteIAMPolicy method.

This doesn't always work, but I think that abandoning the resource in every case is the wrong action to take. For example with KMS crypto key, we attempt to delete all versions belonging to that key before abandoning the key itself.

Reply:
Does this imply that the DCL should have a delete method for every resource?

Reply:
That's how I imagine this working. I would expect the DCL to have a delete method that attempts to clear out the resource in whatever way makes sense for that resource, even if that means abandoning it entirely. This would allow us to hide some of the implementation details of which resources have delete from the Terraform provider, giving us a standard implementation where we just call delete on everything.

Reply:
In that case I will add a delete method to vertex endpoint traffic split that zeros out the field.